pkg/machines/components/graphicsConsole.jsx

```diff
@@ -0,0 +1,198 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from "react";
+import cockpit from 'cockpit';
+import Vnc from './vnc.jsx';
+import DesktopConsoleDownload from './desktopConsole.jsx';
+import { vmDesktopConsole } from '../actions.es6';
+import { logDebug } from '../helpers.es6';
+
+const _ = cockpit.gettext;
+
+const NoConsole = () => { // TODO: add option to define graphics consoles
+    return (
+        <div>
+            {_("No graphics console is defined for this virtual machine.")}
+        </div>
+    );
+};
+
+const VmNotRunning = () => {
+  return (
+      <div>
+          {_("Please start the virtual machine to access its graphics console.")}
```

Comment:
Not a recent use case, but it would be nice to have a "solution" here. 
Anyway I will have a look at the flow and prefer this message over no-console state